### PR TITLE
JDL proposed adjustments

### PR DESCRIPTION
**Pull request by JulianLeviston**
Hi @tdammers , I followed your suggestion and made some suggested edits as a PR. Hope that's good! :)

Julian

### CODE REVIEW

File: newcomers-tutorial.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # The Newcomer's Guide To GHC Development
 
 This page is intended to serve as the first stop for those people who say, "I
-want to contribute to GHC, but I don't know quite where to begin." Begin here. 
+want to contribute to GHC, but I don't know quite where to begin." Begin here.
 
-If you have any questions along the way don't hesitate to reach out to the
+If you have questions along the way, don't hesitate to reach out to the
```
Comment (tdammers) [inline newcomers-tutorial.md]: I'd keep the "any", it makes for a nicer flow IMO.

File: newcomers-tutorial.md [hunk 2]
```diff
@@ -20,31 +20,31 @@ The path to a successful merge request looks like this:
 
 ## Installing Prerequisites
 
-GHC builds on a plethora of platforms; check [Setting up your system
+GHC builds many platforms; check [Setting up your system
```
Comment (tdammers) [inline newcomers-tutorial.md]: Need to keep the "on" - GHC does not build the platform itself. The "plethora" part was copied from an older document, and I don't see anything wrong with it, but I have no strong feelings about it either way.

File: newcomers-tutorial.md [hunk 3]
```diff
@@ -20,31 +20,31 @@ The path to a successful merge request looks like this:
...
 for building GHC](building/preparation) for detailed instructions for your
 platform of choice.
 
-A minimal list of prerequisites:
+To build GHC, you'll need both a fairly regular set of build tools as well as a Haskell toolchain. The standard tools are:
 
 - **git**
-- **a typical build toolchain**: autoconf, automake, libtool, make, gcc, g++
+- **a standard build toolchain**: autoconf, automake, libtool, make, gcc, g++
```
Comment (tdammers) [inline newcomers-tutorial.md:30]: Nothing wrong with "typical" IMO, and those things aren't actually a standard (see above).

File: newcomers-tutorial.md [hunk 4]
```diff
@@ -20,31 +20,31 @@ The path to a successful merge request looks like this:
 
 ## Installing Prerequisites
 
-GHC builds on a plethora of platforms; check [Setting up your system
+GHC builds many platforms; check [Setting up your system
 for building GHC](building/preparation) for detailed instructions for your
 platform of choice.
 
-A minimal list of prerequisites:
+To build GHC, you'll need both a fairly regular set of build tools as well as a Haskell toolchain. The standard tools are:
```
Comment (tdammers) [inline newcomers-tutorial.md]: This doesn't combine well with the paragraph before. We have already explained (in the section title, and the preceding paragraph) that you need to do some preparational work; explaining it *again* in this sentence makes it feel out of place. The previous version of this bit may be a bit spartan, and probably deserves being expanded into a full sentence, but this one here has too much fluff for my taste.

I'd also avoid using the word "standard" for things that aren't actually standardized in any way, which is why I used the word "typical" instead.

File: newcomers-tutorial.md [hunk 5]
```diff
@@ -20,31 +20,31 @@ The path to a successful merge request looks like this:
...
 - **ncurses**
 - **xz-utils**
 
-Install these as usual. You will need the development packages for any system
-libraries (typically named `something-dev` on a Unix-like OS distribution).
+Install these as usual. You will need the development packages for system
+libraries (usually named like `git-dev`, `python3-dev`, etc. on Unix-like OS distributions).
 
-And a Haskell toolchain:
+The Haskell toolchain has:
```
Comment (tdammers) [inline newcomers-tutorial.md:40]: "The" is inappropriate here, because we haven't explicitly mentioned that you need a Haskell toolchain. Wording it as "has" is also a bit strange; I would go for "consists of", or maybe "provides". How about: "You will need a Haskell toolchain, providing:"?

File: newcomers-tutorial.md [hunk 6]
```diff
@@ -20,31 +20,31 @@ The path to a successful merge request looks like this:
...
 
 - **GHC**; you need a reasonably new version, we recommend the latest stable
   release, which you can find via the [GHC homepage](https://www.haskell.org/ghc/).
 - **Cabal**; either install through your distro, and then upgrade with `cabal
   install cabal-install`, or follow the steps to install manually outlined on
   [Cabal's Hackage Page](https://hackage.haskell.org/package/cabal-install)
-- **Alex** and **Happy**; use Cabal to install these.
+- **Alex** and **Happy**; use `cabal install alex` and `cabal install happy` to install these.
```
Comment (tdammers) [inline newcomers-tutorial.md]: I left out the exact commands on purpose, because with new-style commands underway, the v1-style commands will be obsolete soon, but v2-style still has a few rough edges that might prove problematic. Explaining this situation would make things too lengthy, while deciding on one or the other would also be problematic; and further, I think it is safe to assume that someone willing to work on GHC is familiar enough with Cabal to figure out how to install two programs with it. The reason I mention Cabal at all in this bullet point is because installing these two from, say, Debian, or even Haskell Platform, will likely get you older versions, which can cause problems down the road.

File: newcomers-tutorial.md [hunk 7]
```diff
@@ -54,20 +54,17 @@ opt to skip this part for the time being.
   `texlive-fonts-recommended`, `fonts-lmodern`, `texlive-latex-recommended`,
   `texlive-latex-extra` - refer to your distro for available packages)
 
-### Required Bureaucracy
+### Required Accounts
```
Comment (tdammers) [inline newcomers-tutorial.md]: This section originally described additional housekeeping, which is why I chose the term "Bureaucracy". But by now, it's really just creating a gitlab account, so let's just name this section "GitLab Account".

File: newcomers-tutorial.md [hunk 8]
```diff
@@ -54,20 +54,17 @@ opt to skip this part for the time being.
...
 
-GHC uses its own gitlab environment, found at
-[gitlab.haskell.org](https://gitlab.haskell.org/). In order to submit merge
-requests, file issues, and participate in code reviews, you will need to create
-an account, or sign in with a GitHub account.
+In order to submit merge requests, file issues, and participate in code reviews, you will need to create a gitlab account, or sign in with GitHub account. GHC uses its own gitlab environment, found at [gitlab.haskell.org](https://gitlab.haskell.org/).
```
Comment (tdammers) [inline newcomers-tutorial.md]: Flipping the ordering of the two sentences makes it weird IMO - first you tell people to create a gitlab account (which, to the unsuspecting user, means "go to gitlab.com and register an account with them"), and *then* you say, "oh, but when I said "gitlab account", I didn't actually mean gitlab as in gitlab.com, but gitlab as in the gitlab instance we run on gitlab.haskell.org."

Maybe the "uses its own gitlab environment" part isn't as nice as it could be though, so perhaps just this:

```GHC uses a GitLab instance at [gitlab.haskell.org] for code hosting, merge requests, issue tracking, and code reviews, so you will need to create an account for that, or sign in with a GitHub account.```

File: newcomers-tutorial.md [hunk 9]
```diff
@@ -54,20 +54,17 @@ opt to skip this part for the time being.
...
 
 ## Getting The Code
 
-Since we will be committing patches, we will need a GitLab fork, so it's best
-to get that out of the way right away. So sign into GitLab, find [the GHC
+Since you will be committing patches, you will need a GitLab fork, so it's best
+to get that out of the way right away. Sign into GitLab, find [the GHC
 project](https://gitlab.haskell.org/ghc/ghc), and fork it.
```
Comment (tdammers) [inline newcomers-tutorial.md]: This could actually use some shortening. How about:

```
We recommend doing your work on a GitLab fork, and it's best to create one right away.
Sign into GitLab, find [the GHC project](...), and fork it.
```

File: newcomers-tutorial.md [hunk 10]
```diff
@@ -54,20 +54,17 @@ opt to skip this part for the time being.
...
 
-Then clone it to your development machine:
+Then, clone it to your development machine:
```
Comment (tdammers) [inline newcomers-tutorial.md]: Comma is unnecessary here and subtly changes the flow in a way that I don't like a lot. Without the comma, it signals that it's a relatively swift action, but the comma suggests a point where a moment of reflection is required. Conceptually, I want to say "Just fork the project on GitLab, then clone the fork. No big deal.", rather than "First, fork the project. This is a lengthy and cumbersome task. Then, when you have fought your way through that, fork the project, but beware the perils and tread carefully."

Then again, it's just a comma, so maybe I shouldn't obsess about it so much.

File: newcomers-tutorial.md [hunk 11]
```diff
@@ -98,7 +95,8 @@ cd ghc/
   on the compiler itself; without it, a release build will be made, which takes
   significantly longer and is less convenient for development.
 
-Now go make yourself some coffee while the build runs.
+Now go make yourself some coffee while the build runs. Don't be too worried if it
+takes around 20 minutes or more for your first build.
```
Comment (tdammers) [inline newcomers-tutorial.md]: I'd like to avoid giving hard numbers here, because build times tend to vary very drastically between systems. On my dev laptop (a mildly outdated i5 with an SSD/HDD hybrid drive), I couldn't dream of making a full from-scratch development build in just 20 minutes, whereas I've seen people with access to more powerful build servers pull off a full build in a couple minutes; and on machines on the lower end of the realistic spectrum, it can take multiple hours. The humorous "make yourself a coffee" suggestion should hopefully be enough of a hint that this is expected to be, and again, we're not dealing with programming newbies here, our target audience has at least some experience *using* GHC, and they will by now be aware of the fact that it's a large, complex, and old codebase, so long build times should not come as a surprise.

File: newcomers-tutorial.md [hunk 12]
```diff
@@ -122,7 +119,7 @@ approaches you used, you may want to use one of the alternative build methods:
 
 Hadrian is GHC's new build system, based on [Shake](https://shakebuild.com/).
 It is pretty stable already and should work just fine, but if you run into
-issues regardless, it may not be your fault.
+issues, it might help to read the below.
```
Comment (tdammers) [inline newcomers-tutorial.md]: My biggest fear here is people following all the instructions, and then running into a Hadrian bug that looks like they did something wrong when they didn't, and then spend many fruitless hours trying to figure out what they did wrong. Not 100% with the original text though, so this may need a bit more thinking.

File: newcomers-tutorial.md [hunk 13]
```diff
@@ -179,14 +176,13 @@ All is fine.
 
 ### Rebuilding
 
-While working on GHC, you will need to rebuild often; however, most of the
-time, it is possible to avoid a full, slow build. Here's a few things you can
-do to speed things up:
+While working on GHC you will need to rebuild often. Most of the time, it is
```
Comment (tdammers) [inline newcomers-tutorial.md]: I'd keep the comma due to inverted sentence structure.

I'd also keep one of "however", "though", or "but" in there, to keep the "this is true, but you can do that" logic in there. What I'm trying to say here is not "We would like to remind you to rebuild often during development, and you can speed this up by doing the following", but rather "Unfortunately, frequent rebuilds are inevitable during development, but you can alleviate some of the pain by speeding things up, and here's how". So the "but" notion is kind of important here.

File: newcomers-tutorial.md [hunk 14]
```diff
@@ -179,14 +176,13 @@ All is fine.
...
+possible to avoid full, slower builds. Here are a few things you can do
+to speed things up:
 
-- Rebuild only the things that you're interested in. For example, to only build
+- Rebuild only what you're interested in. For example, to only build
   GHC itself, say `./hadrian/build.sh stage2:exe:ghc-bin`. See
-  [the Hadrian README](https://gitlab.haskell.org/ghc/ghc/blob/master/hadrian/README.md)
-  for details.
+  [the Hadrian README](https://gitlab.haskell.org/ghc/ghc/blob/master/hadrian/README.md) for details.
```
Comment (tdammers) [inline newcomers-tutorial.md]: Diff noise, suggest putting the line break back.

File: newcomers-tutorial.md [hunk 15]
```diff
@@ -246,42 +240,39 @@ can specify multiple tests, separated by whitespace:
...
+
+Of course, we can't be sure how hard a task is before doing it, so
+apologies in advance if one of these is too hard.
 
 Either way, **if you are going to work on something, make sure a ticket exists
 for it**. This is essential for all sorts of things, but most importantly,
 attaching things to tickets makes sure they don't get lost. So if there is no
-ticket for it yet, do file one. The issue tracker is [right
-here](https://gitlab.haskell.org/ghc/ghc/issues/new).
+ticket for it yet, please use the [issue tracker](https://gitlab.haskell.org/ghc/ghc/issues/new) to do so.
```
Comment (tdammers) [inline newcomers-tutorial.md:250]: Good, but keep line lengths consistent if possible, and ditch the "to do so" part, or maybe reword as "..., please create one on the [issue tracker](...)".

File: newcomers-tutorial.md [hunk 16]
```diff
@@ -246,42 +240,39 @@ can specify multiple tests, separated by whitespace:
...
 
-Apart from that, you are encouraged to ask for a starting point on IRC or the
-`ghc-devs` [mailing list](mailing-lists-and-irc). There someone familiar with
-the process can help you find a ticket that matches your expertise and help you
-when you get stuck.
+We'd like to encouraged you to ask for a starting point on IRC or the
```
Comment (tdammers) [inline newcomers-tutorial.md]: - It's either "you are encouraged", or "we'd like to encourage you".
- I'd like to keep a notion of "also" in here somehow.
- Maybe "invite" is a better word than "encourage" here.

File: newcomers-tutorial.md [hunk 17]
```diff
@@ -246,42 +240,39 @@ can specify multiple tests, separated by whitespace:
...
+`ghc-devs` [mailing list](mailing-lists-and-irc). You'll find someone familiar
+with the process who can help you find a ticket that matches your expertise and you'll also find people who can help you when you get stuck.
```
Comment (tdammers) [inline newcomers-tutorial.md]: This sentence has become too long / clumsy. At the very least, insert a comma after "expertise"; repeating the "you'll find" part isn't very nice stylistically either. So how about: "You'll find someone familiar with the process who can help you find a ticket that matches your expertise, as well as people who can help you when you get stuck."

File: newcomers-tutorial.md [hunk 18]
```diff
@@ -246,42 +240,39 @@ can specify multiple tests, separated by whitespace:
...
+enforce any in particular. The general rule is to stick to the same coding
 style as is already used in the file you're editing. If you must make stylistic
-changes, commit them separately from functional changes, so that someone
-looking back through the change logs can easily distinguish them.
+changes, commit them separately from functional changes, so someone looking
+back through the change logs can easily distinguish them.
 
-It's much better to write code that is transparent than to write code that is
-short.
+In general, it's much better to write code that is transparent than to write code that is short.
```
Comment (tdammers) [inline newcomers-tutorial.md]: - "In general" is practically empty fluff, doesn't add much value. I'd keep the original.
- Diff noise, suggest putting line break back.

File: newcomers-tutorial.md [hunk 19]
```diff
@@ -246,42 +240,39 @@ can specify multiple tests, separated by whitespace:
...
 #### General
 
 GHC doesn't follow a uniform coding style across all modules, and we do not
-enforce any particular one. The general rule is to stick to the same coding
+enforce any in particular. The general rule is to stick to the same coding
 style as is already used in the file you're editing. If you must make stylistic
-changes, commit them separately from functional changes, so that someone
-looking back through the change logs can easily distinguish them.
+changes, commit them separately from functional changes, so someone looking
+back through the change logs can easily distinguish them.
```
Comment (tdammers) [inline newcomers-tutorial.md:268]: These don't significantly improve things IMO, matter of arbitrary taste. "we do not enforce any in particular" is a bit strange, lacking some sort of noun - should be something like "we do not enforce any one in particular", but that's no better than the original, is it.

File: newcomers-tutorial.md [hunk 20]
```diff
@@ -246,42 +240,39 @@ can specify multiple tests, separated by whitespace:
...
+We'd like to encouraged you to ask for a starting point on IRC or the
+`ghc-devs` [mailing list](mailing-lists-and-irc). You'll find someone familiar
+with the process who can help you find a ticket that matches your expertise and you'll also find people who can help you when you get stuck.
 
 ## Working with the code
 
-GHC is an old codebase, with plenty of historic quirks to it, and the things it
-does are intrinsically complex, so it can be quite overwhelming. Don't let that
-discourage you, and don't hesitate to ask questions.
+GHC is a decades-old codebase, with plenty of historic quirks to it, and the things it does are intrinsically complex, so it can be quite overwhelming.
```
Comment (tdammers) [inline newcomers-tutorial.md]: - "Decades" doesn't anything important, let's keep it short.
- Try to keep line lengths consistent and avoid diff noise
- "Please" seems appropriate, but doesn't have to go on a separate paragraph (which it doesn't anyway, but it looks like that may have been your intention)

File: newcomers-tutorial.md [hunk 21]
```diff
@@ -319,7 +310,7 @@ and lead to further optimisation.
 -}
 ```
 
-Note that:
+Please pay attention to:
```
Comment (tdammers) [inline newcomers-tutorial.md]: This doesn't grammar: "Please pay attention to the Note itself is a long block of prose." Besides, I think "Note that" is perfectly appropriate - we're simply explaining the convention here, and "note that:" is a fairly common way of directing the reader's attention to some points we want to highlight. I don't see a reason to change that.

**Review by tdammers — CHANGES_REQUESTED** [review]
Thanks for you effort, some great stuff in there. I tried to explain some decisions of mine that you may have been unaware of, so by all means consider my remarks constructive criticism.

File: newcomers-tutorial.md (hunk 1 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md]: When editing, I often try to retain the original meaning but remove as much as possible. Doing this invariably makes for clearer reading.

File: newcomers-tutorial.md (hunk 2 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md]: Oops. Thanks. My reasoning was it's often good to choose simpler language as there's less chance of isolating our potential audience.

File: newcomers-tutorial.md (hunk 3 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md:30]: They're standard in the sense of usual or typical, irrespective of the fact that they're not *a* standard. They're the non-haskell parts, so in the sense of regular / normal.

File: newcomers-tutorial.md (hunk 5 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md:40]: I have explicitly mentioned a haskell toolchain - in the paragraph above both the typical and haskell toolchains.

The wording I'm after here is "includes", I think. The meaning is "here are the tools in the toolchain", but with less words. 

In the introductory paragraph ("You'll need both a typical set of build tools as well as a Haskell toolchain.") I'm attempting to smoothly connect the flow from heading up to individual instructions. I find the lack of links and things like "install these as usual" quite jarring, because they seem less helpful than the previous instructions were (at least, as far as my memory goes).

File: newcomers-tutorial.md (hunk 6 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md]: I don't really understand most of what you just said, (I have no idea what v1 or v2 style commands are) and I personally happen to be someone who's new, and not overly familiar with exactly what to type to install things with cabal (I've mostly used stack)... so maybe the assumption everyone knows how to use cabal isn't so good?

Aside from this, I think what I did was follow the instructions for installing these things from their project pages themselves. Maybe we should mention doing that. The thing I'm after here is *smoothness*... it really sucks when one doesn't know where to go to look for things if one gets stuck, and continually asking for help (especially as a beginner doing what seems like the very first thing) is a real downer.

File: newcomers-tutorial.md (hunk 7 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md]: Okay, if that's better. As long as it's not bureaucracy ;-) (reasoning is it's better to use simpler words where possible, unless it has a particular stylistic intent).

File: newcomers-tutorial.md (hunk 8 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md]: Ah. Very interesting... the original intent wasn't overly clear to me! 😁 which is probably why I flipped it easily... because I didn't realise we had our own specific instance.

File: newcomers-tutorial.md (hunk 10 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md]: I was trying to improve the flow. Starting a new paragraph with `Then` is kind of stilted. I was also trying not to rework your writing too much! :) (I prefer not to rework someone's writing too much when I'm editing, because it's their writing I'm editing).

I'll have a think and see if I can come up with something that meets us both half way :)

Comment (JulianLeviston) [reply newcomers-tutorial.md]: Update: okay, I significantly reduced the pargraph. Hopefully I haven't offended you too much with my edits.

File: newcomers-tutorial.md (hunk 11 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md]: Yeah. I realised that when I wrote it, but I was thinking of the poor soul (such as myself on running my first build) who would be worried that something was wrong, and stop the process, then restart it several times, only to find out that 30 minutes for the first build isn't unheard of or crazy. What are your suggestions for that? That's why I said don't be too worried. It's not a hard number, it's a soft number.

I guess this kind of comes down to... what kind of people are we onboarding? Are they 20 year haskell devs, 10 year haskell devs? 2 year haskell devs? 2 month haskell devs? My assumption for new people is we usually want to make it as smooth as possible for as many people who want to contribute as can... because the more people we get who can contribute the merrier. But maybe that's not the general way things are? I don't know. I'm new to this :)

Comment (JulianLeviston) [reply newcomers-tutorial.md]: I'll adjust so it says "if it takes a while". It just is a bit vague... and my personal definition of 'a while' is a lot less time than 20 minutes. I was a bit shocked when it took over 30 minutes for my first build, TBH. (But that's because I didn't realise how *big* haskell was yet then)

File: newcomers-tutorial.md (hunk 12 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md]: That's fine. I took out "it may not be your fault" because it sounds like the default is to blame them! (in other words, it could be construed as slightly aggressive language). One of my favourite things about the original that this text is replacing is that it had a kind of fire-side chat feel to it. It was quite friendly language.

Maybe I'll change it to "if you run into issue, please don't assume it's your own fault."

Comment (JulianLeviston) [reply newcomers-tutorial.md]: Update: okay I reworked it a bit, but I'm not *super* happy with the result yet.

File: newcomers-tutorial.md (hunk 13 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md]: None of those semantics are lost by the rework, but it *does* result in a sentence that flow better (IMO). I'll see if I can meet us both halfway somehow.

Comment (JulianLeviston) [reply newcomers-tutorial.md]: Okay I reflowed it and reduce it. Hopefully you like that more.

File: newcomers-tutorial.md (hunk 16 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md]: Haha oops. Spello :)

Comment (JulianLeviston) [reply newcomers-tutorial.md]: Ok i've reworked this. Think I'm getting a bit tired now, though, so might have to finish soon. I think it's a long way from being really really good yet, but it's probably becoming more than good enough for now :)

File: newcomers-tutorial.md (hunk 17 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md]: Ok. I adjusted it. Feel free to throw more adjustments my way, and maybe once I've slept on it I'll have some better adjustments to make, too :)

File: newcomers-tutorial.md (hunk 21 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md]: Sure. I was trying to get rid of "note note" :-)

**Comment by JulianLeviston** [comment]
@tdammers see what you think of these adjustments. I might make another pass at it if you're not in an extreme rush to get it all merged and stuff. I don't know if you want to take my PR as a set of suggested edits, or you're planning on merging them into *your* PR... either way, I'm happy to have had the discussion. It's interesting. I think we can probably still polish it up a bit more. It's good to get this stuff nice for new people IMO, because it's kind of the first contact they have with our docs on how to begin. ❤️

**Comment by tdammers** [comment]
I think the easiest way is for you to keep amending this PR, and when we're both happy, I'll merge into my PR (which is based on the wip/newcomers branch). And then when the ongoing trac migration effort is through, I can either merge or cherry-pick that into the real repo on gitlab.haskell.org, or, worst case, manually add the page. I'll take a look at your adjustments in a minute.

File: newcomers-tutorial.md (hunk 19 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md:268]: Made a fix.

File: newcomers-tutorial.md (hunk 21 above)
Comment (JulianLeviston) [reply newcomers-tutorial.md]: BTW "this doesn't grammar" is great :)

**Comment by JulianLeviston** [comment]
@tdammers ok, that works. Thanks. :)

**Comment by tdammers** [comment]
OK, so now that the actual wiki migration from Trac to GitLab has run, we decided at GHC HQ that we should merge the guide into the actual wiki in its current state (i.e., without the changes we have been discussing in this PR); you can find it [here](https://gitlab.haskell.org/ghc/ghc/wikis/newcomers-tutorial). I am going to leave the rest of the page's lifecycle up to the community, so all means feel free to edit the live wiki page on gitlab. Since we haven't fully resolved the discussion on this PR however, I will close the PR without merging. That doesn't mean I'm rejecting your suggestions, but rather just that it would be easier for me (and also more transparent to the community) if you could just apply the final set directly to the version on GitLab. If in doubt, we can of course always refer to this PR in the future.

**Comment by JulianLeviston** [comment]
@tdammers apologies for asking here, but can you direct me as to how to make changes to wiki pages on gitlab? I don't appear to be able to do that, and I can't seem to find a wiki entry on how to edit the wiki! :)

**Comment by tdammers** [comment]
Haha, yeah, this is unfortunately slightly in flux right now. The idea is
that all you should need to edit the wiki is an account on
gitlab.haskell.org, but it seems that stock gitlab requires
additional permissions ("developer", IIRC). We're working on it, so
hopefully this will be sorted out soon. Sorry for the inconvenience.

In the meantime, if you like, you can commit to this PR, and I'll see about
getting it in.
